lib/charms/nginx_ingress_integrator/v0/ingress.py

```diff
@@ -189,40 +250,77 @@ def _on_relation_changed(self, event):
+
         # Create an event that our charm can use to decide it's okay to
         # configure the ingress.
         self.charm.on.ingress_available.emit()
 
+    def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
+        """Handle a relation-broken event in the ingress relation."""
+        if not self.model.unit.is_leader():
+            return
 
-class IngressProvides(IngressBaseProvides):
-    """This class defines the functionality for the 'provides' side of the 'ingress' relation.
+        # Create an event that our charm can use to remove the ingress resource.
+        self.charm.on.ingress_broken.emit(event.relation)
 
-    Hook events observed:
-        - relation-changed
-    """
+class IngressProxyProvides(Object):
 
     def __init__(self, charm):
-        super().__init__(charm, "ingress")
+        super().__init__(charm, "ingress-proxy")
         # Observe the relation-changed hook event and bind
         # self.on_relation_changed() to handle the event.
-        self.framework.observe(charm.on["ingress"].relation_changed, self._on_relation_changed)
+        self.framework.observe(charm.on["ingress-proxy"].relation_changed, self._on_relation_changed)
         self.charm = charm
 
+    def _on_relation_changed(self, event):
+        """Handle a change to the ingress relation.
 
-class IngressProxyProvides(IngressBaseProvides):
+        Confirm we have the fields we expect to receive."""
+        # `self.unit` isn't available here, so use `self.model.unit`.
+        if not self.model.unit.is_leader():
+            return
+        
+        if not event.relation.data[event.app]:
+            LOGGER.info("Ingress-proxy hasn't finished configuring, waiting until relation is changed again.")
+            self.charm.on.ingress_proxy_available.emit()
+            return
 
-    def __init__(self, charm):
-        super().__init__(charm, "ingress")
-        # Observe the relation-changed hook event and bind
-        # self.on_relation_changed() to handle the event.
-        self.framework.observe(charm.on["ingress"].relation_changed, self._on_relation_changed)
-        self.charm = charm
+        ingress_data = {
+            field: event.relation.data[event.app].get(field)
+            for field in REQUIRED_INGRESS_RELATION_FIELDS | OPTIONAL_INGRESS_RELATION_FIELDS
+        }
+
+        missing_fields = sorted(
+            [
+                field
+                for field in REQUIRED_INGRESS_RELATION_FIELDS
+                if ingress_data.get(field) is None
+            ]
+        )
+
+        if missing_fields:
+            LOGGER.error(
+                "Missing required data fields for ingress-proxy relation: {}".format(
+                    ", ".join(missing_fields)
+                )
+            )
+            self.model.unit.status = BlockedStatus(
+                "Missing fields for ingress-proxy: {}".format(", ".join(missing_fields))
+            )
+            self.charm.on.ingress_available.emit()
```

Comment:
It seems odd to me that we're emitting a ingress event here, not an ingress_proxy event. Are we sure that's what we want?